wee_slack.py

```diff
@@ -1386,142 +1415,148 @@ def create_reaction_string(reactions):
         reaction_string = ''
     return reaction_string
 
-# deprecated in favor of redrawing the entire buffer
-#def modify_buffer_line(buffer, user, new_message, time, append):
-#    time = int(float(time))
-#    own_lines = w.hdata_pointer(w.hdata_get('buffer'), buffer, 'own_lines')
-#    if own_lines:
-#        line = w.hdata_pointer(w.hdata_get('lines'), own_lines, 'last_line')
-#        hdata_line = w.hdata_get('line')
-#        hdata_line_data = w.hdata_get('line_data')
-#
-#        while line:
-#            data = w.hdata_pointer(hdata_line, line, 'data')
-#            if data:
-#                date = w.hdata_time(hdata_line_data, data, 'date')
-#                prefix = w.hdata_string(hdata_line_data, data, 'prefix')
-#                if new_message == "":
-#                    new_message = w.hdata_string(hdata_line_data, data, 'message')
-#                if user and (int(date) == int(time) and user == prefix):
-#                    w.prnt("", "found matching time date is {}, time is {} ".format(date, time))
-#                    w.hdata_update(hdata_line_data, data, {"message": "{}{}".format(new_message, append)})
-#                    break
-#                elif not user and (int(date) == int(time)):
-#                    w.hdata_update(hdata_line_data, data, {"message": "{}{}".format(new_message, append)})
-#                else:
-#                    pass
-#            line = w.hdata_move(hdata_line, line, -1)
-#    return w.WEECHAT_RC_OK
-
+def modify_buffer_line(buffer, new_line, time):
```

Comment:
Return to more efficient handling of changes: edits, deletes, emoji reactions, etc are now done by modifying lines instead of redrawing the buffer completely.